spec/reek/tree_walker_spec.rb

```diff
@@ -1,237 +1,238 @@
+      end
 
-  it 'counts 1 nested assignment' do
-    method = process_method('def one() val = fred = 4; end')
-    expect(method.num_statements).to eq(1)
-  end
+      it 'has one child' do
+        expect(context_tree.children.size).to eq(1)
+      end
 
-  it 'counts returns' do
-    method = process_method('def one() val = 4; true; end')
-    expect(method.num_statements).to eq(2)
-  end
+      describe 'the root node' do
+        let(:module_context) { context_tree.children.first }
 
-  it 'counts nil returns' do
-    method = process_method('def one() val = 4; nil; end')
-    expect(method.num_statements).to eq(2)
-  end
+        it 'has one module_context' do
+          expect(module_context).to be_a(Reek::Context::ModuleContext)
+        end
 
-  context 'with control statements' do
-    it 'counts 1 statement in a conditional expression' do
-      method = process_method('def one() if val == 4; callee(); end; end')
-      expect(method.num_statements).to eq(1)
-    end
+        it 'holds a reference to the parent context' do
+          expect(module_context.send(:context)).to eq(context_tree)
+        end
 
-    it 'counts 3 statements in a conditional expression' do
-      method = process_method('def one() if val == 4; callee(); callee(); callee(); end; end')
-      expect(method.num_statements).to eq(3)
-    end
+        describe 'the module node' do
+          let(:method_context) { module_context.children.first }
 
-    it 'counts 1 statements in an else' do
-      method = process_method('def one() if val == 4; callee(); else; callee(); end; end')
-      expect(method.num_statements).to eq(2)
-    end
+          it 'has one method_context' do
+            expect(method_context).to be_a(Reek::Context::MethodContext)
+            expect(module_context.children.size).to eq(1)
+          end
 
-    it 'counts 3 statements in an else' do
-      method = process_method('
-        def one()
-          if val == 4
-            callee(); callee(); callee()
-          else
-            callee(); callee(); callee()
+          it 'holds a reference to the parent context' do
+            expect(method_context.send(:context)).to eq(module_context)
           end
         end
-                              ')
-      expect(method.num_statements).to eq(6)
-    end
-
-    it 'does not count constant assignment with or equals' do
-      source = 'class Hi; CONST ||= 1; end'
-      klass = process_method(source)
-      expect(klass.num_statements).to eq(0)
-    end
-
-    it 'does not count multi constant assignment' do
-      source = 'class Hi; CONST, OTHER_CONST = 1, 2; end'
-      klass = process_method(source)
-      expect(klass.num_statements).to eq(0)
+      end
     end
+  end
 
-    it 'does not count empty conditional expression' do
-      method = process_method('def one() if val == 4; ; end; end')
-      expect(method.num_statements).to eq(0)
+  describe 'statement counting' do
```

Comment:
I more or less left those specs unchanged. I just simplified them in regards to their test set up and also removed a couple of more or less duplicates.